File: 20-ValidParentheses/20-ValidParentheses.py
# Last updated: 6/1/2025, 11:24:37 PM
class Solution(object):
    def isValid(self, s):
        stack = []
        for c in s:
            if c in '([{':
                stack.append(c)
            else:
                if not stack or (c == ')' and stack[-1] != '(') or (c == '}' and stack[-1] != '{') or (c == ']' and stack[-1] != '['):
                    return False
                stack.pop() 
        return not stack

File: 20-ValidParentheses/20-ValidParentheses.java
// Last updated: 6/3/2025, 12:01:45 AM
class Solution {
    public int search(int[] nums, int target) {
        int index = -1;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == target){
                index = i;
                break;
            }
        }
        return index;
    }
}